sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java

```diff
@@ -486,7 +486,7 @@ public void populateDisplayData(DisplayData.Builder builder) {
           // We first apply a (trivial) transform to the input PCollection to produce a new
           // PCollection. This ensures that we don't modify the windowing strategy of the input
           // which may be used elsewhere.
-          .apply("Identity", MapElements.via(new SimpleFunction<T, T>() {
+          .apply("Identity", MapElements.<T,T>via(new SimpleFunction<T, T>() {
```

Comment:
For what it's worth, when I switch to language level = 7 at least one other different file `CoderRegistry.java` starts having an error:

![screenshot from 2017-04-25 21-19-27](https://cloud.githubusercontent.com/assets/35872/25422593/134bf83e-29fd-11e7-8311-81ca9a63c991.png)